### ISSUE TITLE
MHLO CAPI is not actually a CAPI

### ISSUE THREAD

**Issue by stellaraccident** [tensorflow/mlir-hlo#33]
Example:

```
In file included from /home/stella/src/iree/third_party/mlir-hlo/python/MlirHloModule.cpp:15:
/home/stella/src/iree/third_party/mlir-hlo/include/mlir-hlo-c/Attributes.h:186:32: warning: 'mlirMhloComparisonDirectionAttrGetDirection' has C-linkage specified, but returns user-defined type 'std::string' (aka 'basic_string<char>') which is incompatible with C [-Wreturn-type-c-linkage]
MLIR_CAPI_EXPORTED std::string mlirMhloComparisonDirectionAttrGetDirection(
```

Suggest adding a dummy `.c` test which at least includes all of the public headers and will hard fail on such situations.

**Comment by silvasean** [comment]
+1, hitting this in my build too.

There are many functions like that so it results in a wall of build spew.

**Comment by stellaraccident** [comment]
Any update on this? In certain contexts this is illegal, and in others, it spews warnings. I think that we should delete these offending functions, which are clearly incorrect.

**Comment by burmako** [comment]
Thank you, Stella! We'll look into this shortly.

**Comment by burmako** [comment]
Status update: @lipracer has kindly fixed the problems in https://github.com/tensorflow/tensorflow/pull/55888, and we landed the fix in yesterday in 9d95d60963792d0d2ac9973a71c4e1ccc3e28a8c. Thank you so much!

To prevent similar issues in the future, we'll add a dummy `.c` test and connect it to CI to make sure that the CAPI actually stays a CAPI.
